lib/Analysis/Allocation.cpp

```diff
@@ -277,6 +278,15 @@ class AllocationAnalysis {
     // Get the alloc values
     operation->walk<WalkOrder::PreOrder>([&](Operation *op) {
       getExplicitValueSize(op);
+      if (scratchSizeGetter) {
+        constexpr size_t scratchAlignment = 128;
+        unsigned bytes = scratchSizeGetter(op);
+        if (bytes != invalidAllocationSize) {
+          maybeAddScratchBuffer<BufferT::BufferKind::Scratch>(op, bytes,
+                                                              scratchAlignment);
+          return;
+        }
+      }
       getScratchValueSize(op);
```

Comment:
Why not just making this function a parameter?
`getScratchValueSize` itself is also a `scratchSizeGetter` so it can be the default method.

Reply:
Good point! I did that and I'd say all looks clearer now. Thanks!